selfdrive/car/hyundai/hyundaicanfd.py

```diff
@@ -39,13 +39,22 @@ def create_cam_0x2a4(packer, camera_values):
   })
   return packer.make_can_msg("CAM_0x2a4", 4, camera_values)
 
-def create_buttons(packer, cnt, btn):
+def create_buttons(packer, CP, cnt, btn):
+
+  ret = []
+
   values = {
     "COUNTER": cnt,
     "SET_ME_1": 1,
     "CRUISE_BUTTONS": btn,
   }
-  return packer.make_can_msg("CRUISE_BUTTONS", 5, values)
+  
+  if CP.flags & HyundaiFlags.CANFD_HDA2:
+    ret.append(packer.make_can_msg("CRUISE_BUTTONS", 5, values))
+  else:
+    ret.append(packer.make_can_msg("CRUISE_BUTTONS", 6, values))
+  
+  return ret 
```

Comment:
```suggestion
  if CP.flags & HyundaiFlags.CANFD_HDA2:
    return packer.make_can_msg("CRUISE_BUTTONS", 5, values)
  else:
    return packer.make_can_msg("CRUISE_BUTTONS", 6, values)
```

no need to make it a list, I'd also see if we might want a helper function to identify this bus like we do for e can

Reply:
My code didn't work anyways... It gives an error. I will try your code and see if that works for me.